miden-lib/asm/miden/kernels/tx/prologue.masm

```diff
@@ -384,6 +391,54 @@ proc.validate_new_account
     # => []
 end
 
+#! Validates that account procedures match account code commitment and saves procedure information
+#! into memory.
+#!
+#! This is achieved by reading account procedure information from the advice map, saving it into
+#! memory, and sequentially hashing them before comparing the final hash to the account code
+#! commitment.
+#!
+#! Information for each saved procedure consists of the procedure's MAST root and the storage
+#! offset associated with this procedure.
+#!
+#! Stack: [CODE_COMMITMENT]
+#! Output: []
+proc.validate_account_procedures
+  # move procedure data from the advice map to the advice stack and then push the number of
+  # procedures onto the operand stack before storing it in memory
+  adv.push_mapval adv_push.1
+  # => [num_procs, CODE_COMMITMENT]
+
+  # assert that account does not exceed allowed maximum number of procedures
+  # and store number of procedures in memory
+  dup exec.account::get_max_num_procedures lt assert.err=ERR_ACCT_TOO_MANY_PROCEDURES dup exec.memory::set_num_account_procedures
+  # => [num_procs, CODE_COMMITMENT]
```

Comment:
nit: i would split the code into two lines. Maybe something like this:
```
dup exec.account::get_max_num_procedures lt assert.err=ERR_ACCT_TOO_MANY_PROCEDURES
dup exec.memory::set_num_account_procedures
```

Reply:
Split in two separate lines.